src/test/scala/chiselTests/experimental/hierarchy/InstantiateSpec.scala

```diff
@@ -144,6 +145,17 @@ object InstantiateSpec {
   }
 }
 
+@instantiable
+class ParameterizedReset(hasAsyncNotSyncReset: Boolean) extends Module {
+  override def resetType = if (hasAsyncNotSyncReset) Module.ResetType.Asynchronous else Module.ResetType.Synchronous
+  @public val in = IO(Input(UInt(8.W)))
+  @public val out = IO(Output(UInt(8.W)))
+  @public val reset = reset
+  val reg = RegInit(0.U)
+  reg := in
+  out := reg
+}
```

Comment:
yeah, i was eybeballign the verilog in another much noisier version, was also worried that it would constant prop everything away, but i guess Convert does not do that anyway